tools/clone_nonbillables_and_process_invoice.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+set -xe
+
+# Add deploy key to ssh config
+mkdir -p ~/.ssh
+if [ ! -e ~/.ssh/config ]; then
+    touch ~/.ssh/config
+    touch ~/.ssh/id_nonbillable
+    echo "
+    Host github-nonbillable
+        HostName github.com
+        IdentityFile ~/.ssh/id_nonbillable
+    " > ~/.ssh/config
+    echo "$GH_NONBILLABLE_DEPLOYKEY" > ~/.ssh/id_nonbillable
+    chmod 600 ~/.ssh/id_nonbillable
+fi
+
+if [ ! -d ./non-billable-projects ]; then
+    touch ~/.ssh/known_hosts
```

Comment:
Don't hardcode the public keys for GitHub. Make the script auto accept.

Reply:
Fair point, this would harden the security at the cost of a bit more complexity. GitHub advertises this at https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints which is server over HTTPS. You can leave them in, but limit it to only if the `.known_hosts` file doesn't exist, otherwise re-running the command will append them again.

Reply:
I re-added the public keys again.